.github/workflows/release.yml

```diff
@@ -5,40 +5,81 @@ on:
       - 'documentation/**'
     tags:
       - "v1.*"
-
 name: Release
-
 concurrency:
   group: ${{ github.workflow }}-${{ github.ref }}
   cancel-in-progress: true
 
 jobs:
   # ------------------------------------
-  # 1) Build CLI for multiple OS/Arch
+  # 1) Set version variables first
+  # ------------------------------------
+  prepare-version:
+    name: Prepare Version
+    runs-on: ubuntu-latest
+    outputs:
+      version: ${{ steps.set-version.outputs.version }}
+    steps:
+      - name: Extract version
+        id: set-version
+        run: |
+          VERSION=${GITHUB_REF#refs/tags/v}
+          echo "version=$VERSION" >> $GITHUB_OUTPUT
+
+  # ------------------------------------
+  # 2) Update version in files
+  # ------------------------------------
+  update-version:
+    name: Update Cargo.toml & package.json Version
+    needs: [prepare-version]
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+      - name: Update workspace Cargo.toml version
+        run: sed -i 's/^version = ".*"/version = "'${{ needs.prepare-version.outputs.version }}'"/' Cargo.toml
+      - name: Update package.json version
+        run: |
+          npm version ${{ needs.prepare-version.outputs.version }} --no-git-tag-version --allow-same-version
+        working-directory: ui/desktop
+      - name: Commit updated files
+        run: |
+          git config --global user.name "github-actions"
+          git config --global user.email "github-actions@github.com"
+          git commit -am "Update versions to ${{ needs.prepare-version.outputs.version }}"
```

Comment:
nit: should we be more explicit and select the exact files we're committing (`Cargo.toml` and `ui/desktop/package.json`) just in case other things change